### DISCUSSION TITLE
Version 6.0.2

### DISCUSSION THREAD

**Discussion by IvyZX** [google/flax#2615] [General]
## What's Changed
* Refactor out dataclass transform that allows `parent` and `name` to be moved to the end of the argument list into a more general "kw_only_dataclasses" module. by @copybara-service in https://github.com/google/flax/pull/2468
* Don't create reference cycles among Modules. by @levskaya in https://github.com/google/flax/pull/2499
* adds perturb to docs by @cgarciae in https://github.com/google/flax/pull/2511
* Add pre-commit hook to remove trailing white spaces by @cgarciae in https://github.com/google/flax/pull/2513
* Add extracting gradients section to capture intermediates guide by @cgarciae in https://github.com/google/flax/pull/2515
* Make `is_fully_replicated` and `is_fully_addressble` a property rather than a method. by @copybara-service in https://github.com/google/flax/pull/2516
* Adds CallSetupUnboundModuleError by @cgarciae in https://github.com/google/flax/pull/2496
* Adding documentation to Dropout around rng use by @zaxtax in https://github.com/google/flax/pull/2492
* no-op when double wrapping with struct.dataclass by @cgarciae in https://github.com/google/flax/pull/2505
* Add epub and pdf RDT formats by @cgarciae in https://github.com/google/flax/pull/2517
* Update landing page example by @cgarciae in https://github.com/google/flax/pull/2366
* Use `gfile.remove` for files because it doesn't work on GCS files. by @IvyZX in https://github.com/google/flax/pull/2518
* Add save_checkpoint_multiprocess to api reference. by @IvyZX in https://github.com/google/flax/pull/2519
* Add Tensorstore back to required dependencies. by @IvyZX in https://github.com/google/flax/pull/2520
* Delete flax.optim.rst by @ppwwyyxx in https://github.com/google/flax/pull/2522
* Added an error for when we call init, init_with_output and apply on Module class. by @chiamp in https://github.com/google/flax/pull/2529
* Adding link to Bayesian Inference example that uses Flax by @zaxtax in https://github.com/google/flax/pull/2521
* Added IncorrectPostInitOverrideError to capture incorrect post init overrides. by @copybara-service in https://github.com/google/flax/pull/2535
* Fix to use `is_initializing` for init-detection. by @yotarok in https://github.com/google/flax/pull/2486
* add rng_collection argument to Dropout by @cgarciae in https://github.com/google/flax/pull/2540
* Cancel tests if other jobs fail by @cgarciae in https://github.com/google/flax/pull/2507
* Update Guides link in Flax README by @8bitmp3 in https://github.com/google/flax/pull/2544
* Pin jupytext version in requirements.txt by @IvyZX in https://github.com/google/flax/pull/2545
* Fix flax.linen.stochastic.Dropout by @dslisleedh in https://github.com/google/flax/pull/2510
* Transfer Learning Guide by @cgarciae in https://github.com/google/flax/pull/2394
* Update Flax Contributing.md by @8bitmp3 in https://github.com/google/flax/pull/2546
* cap dynamic scale to float32 max by @jheek in https://github.com/google/flax/pull/2553
* Remove optional import of jax.experimental.array for older JAX versions. by @copybara-service in https://github.com/google/flax/pull/2552
* Update examples.rst with denoising-diffusion-flax by @yiyixuxu in https://github.com/google/flax/pull/2487
* return None if no _parent_ref is set by @cgarciae in https://github.com/google/flax/pull/2548
* Add a documentation page on checkpointing by @IvyZX in https://github.com/google/flax/pull/2530
* Lint Flax Contributing guide by @8bitmp3 in https://github.com/google/flax/pull/2560
* Remove extra metadata in Checkpointing guide by @8bitmp3 in https://github.com/google/flax/pull/2559
* Update getting_started.ipynb and getting_started.md by @chiamp in https://github.com/google/flax/pull/2563
* Remove unused svn dependency  by @8bitmp3 in https://github.com/google/flax/pull/2574
* Fix pytype check in checkpoints.py by @IvyZX in https://github.com/google/flax/pull/2592
* Add new 🔪 Flax - The Sharp Bits 🔪 Dropout and randomness by @8bitmp3 in https://github.com/google/flax/pull/2593
* Fixes `launch_gce.sh` with imagenet example. by @andsteing in https://github.com/google/flax/pull/2598
* Added test to check for Variable warning. by @chiamp in https://github.com/google/flax/pull/2610
* Release version 0.6.2 by @IvyZX in https://github.com/google/flax/pull/2613

## New Contributors
* @zaxtax made their first contribution in https://github.com/google/flax/pull/2492
* @ppwwyyxx made their first contribution in https://github.com/google/flax/pull/2522
* @chiamp made their first contribution in https://github.com/google/flax/pull/2529
* @yotarok made their first contribution in https://github.com/google/flax/pull/2486
* @yiyixuxu made their first contribution in https://github.com/google/flax/pull/2487

**Full Changelog**: https://github.com/google/flax/compare/v0.6.1...v0.6.2

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.6.2'>Version 6.0.2</a>.</em>
